src/platform/stub-setup.ts

```diff
@@ -16,6 +17,17 @@ const smarthomeStub: {
   },
 };
 
-export function injectSmarthomeStubs() {
+// Injects stubs into the global scope, allowing implicit usage for testing
+export function injectSmarthomeStubs(): void {
   (global as any).smarthome = smarthomeStub;
-}
\ No newline at end of file
+}
+
+// Promotes App to AppStub
+export function extractMockLocalHomePlatform(
```

Comment:
I wonder if we actually need this, it should be always safe for developer to downcast to `AppStub` within their test.

Reply:
I think this function is okay to keep around for now.  Since the developer would need the `MockLocalHomePlatform` for essentially every test, I think wrapping this cast would save some repetition in the long run.

Reply:
What about calling this `withMockPlatform(app)` and updating the public interface to be fluent?